improver_tests/nbhood/recursive_filter/test_RecursiveFilter.py

```diff
@@ -521,6 +521,53 @@ def test_coordinate_reordering_with_different_smoothing_coefficients(self):
         )
         self.assertArrayAlmostEqual(result.data[0], expected_result)
 
+    def test_mask_zeros(self):
+        """Test that any zeros in the data are the same at the beginning and
+        end of the recursive filtering with the mask_zeros option."""
+        zeros = self.cube.data==0.0
+        plugin = RecursiveFilter(
+            iterations=self.iterations,
+        )
+        result = plugin(
+            self.cube,
+            smoothing_coefficients=self.smoothing_coefficients,
+            mask_zeros=True,
+        )
+        result_zeros = result.data==0.0
+        self.assertArrayEqual(zeros, result_zeros)
+
+    def test_mask_zeros_same_mask(self):
+        """Test that if mask_zeros is applied to a cube that already
+        has a mask, the end result has the same mask."""
+        mask = np.zeros(self.cube.data.shape)
+        mask[0][3][2] = 1
```

Comment:
Just to test the masking with some originally masked zeros and some masked values as well.
```suggestion
        mask[0, 3, :] = 1
```

Reply:
Changed.